stan/math/prim/scal/fun/as_array_or_scalar.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_AS_ARRAY_OR_SCALAR_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_AS_ARRAY_OR_SCALAR_HPP
+
+namespace stan {
+namespace math {
+
+/**
+ * Converts a matrix type to an array. For scalar inputs this is identity
+ * function.
+ *
+ * @tparam T Type of element.
+ * @param v Specified value.
+ * @return Same value.
+ */
+template <typename T>
+inline T as_array_or_scalar(const T& v) {
```

Comment:
Return type should be `const T&` because the scalar type can be `var` or `fvar<var>` in nested autodiff, and we want to pass those by reference.